packages/earth-map/src/pages/verify-email/VerifyEmail.tsx

```diff
@@ -0,0 +1,129 @@
+/*
+  Copyright 2018-2020 National Geographic Society
+
+  Use of this software does not constitute endorsement by National Geographic
+  Society (NGS). The NGS name and NGS logo may not be used for any purpose without
+  written permission from NGS.
+
+  Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+  this file except in compliance with the License. You may obtain a copy of the
+  License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed
+  under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+  CONDITIONS OF ANY KIND, either express or implied. See the License for the
+  specific language governing permissions and limitations under the License.
+*/
+
+import { useAuth0 } from 'auth/auth0';
+import { APP_NAME, ENABLE_PUBLIC_ACCESS } from 'config';
+import React, { useState } from 'react';
+import ProfileService from 'services/ProfileService';
+
+import { Spinner } from '@marapp/earth-shared/src';
+
+import './styles.scss';
+
+export const APP_LOGO = require('images/unbl_logo.svg');
+
+interface EmailState {
+  type: 'inactive' | 'pending' | 'done' | 'error';
+  text: string;
+}
+
+const VerifyEmail = ({ returnToHome }) => {
+  const { logout, isEmailVerified } = useAuth0();
+
+  return (
+    <div className="verified-page marapp-qa-verify-email">
+      <div className="verified-container">
+        {isEmailVerified ? (
+          <VerifiedEmailTemplate returnToHome={returnToHome} />
+        ) : (
+          <VerifyEmailTemplate logout={logout} />
+        )}
+      </div>
+    </div>
+  );
+};
+
+const VerifyEmailTemplate = ({ logout }) => {
+  const [emailState, setEmailState] = useState<EmailState>({
+    type: 'inactive',
+    text: 'Send email verification Link',
+  });
+
+  const handleResendEmail = async () => {
+    setEmailState({ type: 'pending', text: 'sending email validation' });
+
+    try {
+      await ProfileService.resendEmailConfirmation();
+      setEmailState({ type: 'done', text: 'email sent' });
+    } catch (error) {
+      setEmailState({ type: 'error', text: 'error' });
```

Comment:
Provide a better error message, even if it's hardcoded: eg: "Something went wrong. Please try again"